tests/action_tests.py

```diff
@@ -89,6 +102,110 @@ def test_failed_provision_account(get_patch):
+    with no data
+    """
+
+    fake_account_id = "FakeAccount-12345"
+    boompy.set_auth("account_id", "username", "password")
+    boompy.actions.updatePartnerCustomerAccount(fake_account_id, {})
+
+@raises(boompy.errors.BoomiError)
+def test_update_with_missing_id():
+    """
+    Test to see if a BoomiError is thrown when trying to update an account
+    with missing id field
+    """
+
+    fake_account_id = "FakeAccount-12345"
+    fake_data = {
+        "street": "Fake Street"
+    }
+
+    boompy.set_auth("account_id", "username", "password")
+    boompy.actions.updatePartnerCustomerAccount(fake_account_id, fake_data)
+
+@raises(boompy.errors.BoomiError)
+def test_update_with_mismatch():
+    """
+    Test to see if a BoomiError is thrown when trying to update an account
+    when account_id and id field do not match
+    """
+
+    fake_account_id = "FakeAccount-12345"
+    fake_data = {
+        "id": "FakeAccount2-43215",
+        "street": "Fake Street"
+    }
+
+    boompy.set_auth("account_id", "username", "password")
+    boompy.actions.updatePartnerCustomerAccount(fake_account_id, fake_data)
+
+@mock.patch.object(requests.Session, "post")
+def test_successful_update_on_account(post_patch):
+    """
+    Test to simulate a successful account update
+    """
+
+    fake_account_id = "FakeAccount-12345"
+    fake_data = {
+        "id": fake_account_id,
+        "name": "HelloWorld",
+        "street": "Fake Street",
+        "city": "Fake City",
+        "stateCode": "NA",
+        "zipCode": "00000",
+        "countryCode": "NA",
+        "status": "trial",
+        "product": [{"productCode": "fake", "quantity": 1}]
+    }
+
+    fake_content = {
+        "@type": "AccountProvisionResult",
+        "status": "COMPLETED"
+    }
+
+    res_patch = mock.Mock(spec=requests.Response)
+    res_patch.status_code = 200
+    res_patch.content = json.dumps(fake_content)
+    post_patch.return_value = res_patch
+    boompy.set_auth("account_id", "username", "password")
+    boompy.actions.updatePartnerCustomerAccount(fake_account_id, fake_data)
+
+@mock.patch.object(requests.Session, "post")
+@raises(boompy.errors.BoomiError)
+def test_failed_update_on_account(post_patch):
+    """
+    Test to simulate a failed account update
+    """
+
+    fake_account_id = "FakeAccount-12345"
+    fake_data = {
+        "id": fake_account_id,
+        "name": "HelloWorld",
+        "street": "Fake Street",
+        "city": "Fake City",
+        "stateCode": "NA",
+        "zipCode": "00000",
+        "countryCode": "NA",
+        "product": [{"productCode": "fake", "quantity": 1}]
+    }
+
+    fake_content = {
+        "@type": "AccountProvisionResult",
+        "status": "NOT PENDING OR COMPLETED"
+    }
+
+    res_patch = mock.Mock(spec=requests.Response)
+    res_patch.status_code = 200
+    res_patch.content = json.dumps(fake_content)
+    post_patch.return_value = res_patch
```

Comment:
There's a fair amount of boilerplate in making these mock responses... might be worth factoring out a helper for this so you can just do something like `post_patch.return_value = make_mock_json_response(content={"@type": ...})`